### PR TITLE
Added new ebuilds for QtSerialBus package

### PR DESCRIPTION
**Pull request by p2k-ko**
Hi, Qt 5.8 introduced the QtSerialBus package. It is mainly used to connect CAN-bus devices
and other industrial bus types. I've added a new package `dev-qt/qtserialbus` and performed the tests on my Gentoo installation. 

```
>>> Installing (2 of 2) dev-qt/qtserialbus-5.9.1::SchneiderOverlay
 * checking 78 files for package collisions
>>> Merging dev-qt/qtserialbus-5.9.1 to /
--- /usr/
--- /usr/include/
--- /usr/include/qt5/
>>> /usr/include/qt5/QtSerialBus/
>>> /usr/include/qt5/QtSerialBus/QtSerialBusDepends
>>> /usr/include/qt5/QtSerialBus/qmodbustcpserver.h
>>> /usr/include/qt5/QtSerialBus/QModbusTcpClient
>>> /usr/include/qt5/QtSerialBus/qmodbuspdu.h
>>> /usr/include/qt5/QtSerialBus/QModbusExceptionResponse
>>> /usr/include/qt5/QtSerialBus/qcanbusdeviceinfo.h
>>> /usr/include/qt5/QtSerialBus/QModbusRequest
>>> /usr/include/qt5/QtSerialBus/qmodbusreply.h
>>> /usr/include/qt5/QtSerialBus/qmodbusserver.h
>>> /usr/include/qt5/QtSerialBus/qcanbusfactory.h
>>> /usr/include/qt5/QtSerialBus/QModbusRtuSerialMaster
>>> /usr/include/qt5/QtSerialBus/qmodbusrtuserialslave.h
>>> /usr/include/qt5/QtSerialBus/QModbusPdu
>>> /usr/include/qt5/QtSerialBus/qmodbustcpclient.h
>>> /usr/include/qt5/QtSerialBus/qmodbusdevice.h
>>> /usr/include/qt5/QtSerialBus/qserialbusglobal.h
>>> /usr/include/qt5/QtSerialBus/QtSerialBus
>>> /usr/include/qt5/QtSerialBus/QModbusDataUnitMap
>>> /usr/include/qt5/QtSerialBus/qmodbusdataunit.h
>>> /usr/include/qt5/QtSerialBus/QtSerialBusVersion
>>> /usr/include/qt5/QtSerialBus/QCanBus
>>> /usr/include/qt5/QtSerialBus/QModbusRtuSerialSlave
>>> /usr/include/qt5/QtSerialBus/QModbusServer
>>> /usr/include/qt5/QtSerialBus/QModbusDevice
>>> /usr/include/qt5/QtSerialBus/QCanBusFactory
>>> /usr/include/qt5/QtSerialBus/QModbusDataUnit
>>> /usr/include/qt5/QtSerialBus/QCanBusFactoryV2
>>> /usr/include/qt5/QtSerialBus/qcanbus.h
>>> /usr/include/qt5/QtSerialBus/QModbusResponse
>>> /usr/include/qt5/QtSerialBus/QModbusTcpServer
>>> /usr/include/qt5/QtSerialBus/QModbusDeviceIdentification
>>> /usr/include/qt5/QtSerialBus/qtserialbusversion.h
>>> /usr/include/qt5/QtSerialBus/qmodbusclient.h
>>> /usr/include/qt5/QtSerialBus/QCanBusDevice
>>> /usr/include/qt5/QtSerialBus/qcanbusdevice.h
>>> /usr/include/qt5/QtSerialBus/QModbusClient
>>> /usr/include/qt5/QtSerialBus/qmodbusrtuserialmaster.h
>>> /usr/include/qt5/QtSerialBus/QCanBusDeviceInfo
>>> /usr/include/qt5/QtSerialBus/qcanbusframe.h
>>> /usr/include/qt5/QtSerialBus/qtserialbus-config.h
>>> /usr/include/qt5/QtSerialBus/QCanBusFrame
>>> /usr/include/qt5/QtSerialBus/5.9.1/
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/qmodbustcpclient_p.h
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/qcanbusdevice_p.h
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/qmodbus_symbols_p.h
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/qmodbustcpserver_p.h
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/qmodbuscommevent_p.h
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/qmodbusrtuserialmaster_p.h
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/qmodbusrtuserialslave_p.h
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/qtserialbus-config_p.h
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/qmodbusadu_p.h
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/qcanbusdeviceinfo_p.h
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/qmodbusdevice_p.h
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/qmodbusserver_p.h
>>> /usr/include/qt5/QtSerialBus/5.9.1/QtSerialBus/private/qmodbusclient_p.h
>>> /usr/include/qt5/QtSerialBus/qmodbusdeviceidentification.h
>>> /usr/include/qt5/QtSerialBus/QModbusReply
--- /usr/lib64/
--- /usr/lib64/pkgconfig/
>>> /usr/lib64/pkgconfig/Qt5SerialBus.pc
--- /usr/lib64/cmake/
>>> /usr/lib64/cmake/Qt5SerialBus/
>>> /usr/lib64/cmake/Qt5SerialBus/Qt5SerialBusConfigVersion.cmake
>>> /usr/lib64/cmake/Qt5SerialBus/Qt5SerialBus_SocketCanBusPlugin.cmake
>>> /usr/lib64/cmake/Qt5SerialBus/Qt5SerialBus_PeakCanBusPlugin.cmake
>>> /usr/lib64/cmake/Qt5SerialBus/Qt5SerialBus_TinyCanBusPlugin.cmake
>>> /usr/lib64/cmake/Qt5SerialBus/Qt5SerialBusConfig.cmake
--- /usr/lib64/qt5/
--- /usr/lib64/qt5/bin/
>>> /usr/lib64/qt5/bin/canbusutil
--- /usr/lib64/qt5/mkspecs/
--- /usr/lib64/qt5/mkspecs/modules/
>>> /usr/lib64/qt5/mkspecs/modules/qt_lib_serialbus_private.pri
>>> /usr/lib64/qt5/mkspecs/modules/qt_lib_serialbus.pri
--- /usr/lib64/qt5/plugins/
>>> /usr/lib64/qt5/plugins/canbus/
>>> /usr/lib64/qt5/plugins/canbus/libqtsocketcanbus.so
>>> /usr/lib64/qt5/plugins/canbus/libqttinycanbus.so
>>> /usr/lib64/qt5/plugins/canbus/libqtpeakcanbus.so
>>> /usr/lib64/libQt5SerialBus.so.5.9.1
>>> /usr/lib64/libQt5SerialBus.so.5.9 -> libQt5SerialBus.so.5.9.1
>>> /usr/lib64/libQt5SerialBus.prl
--- /usr/lib/
--- /usr/lib/debug/
--- /usr/lib/debug/usr/
--- /usr/lib/debug/usr/lib64/
>>> /usr/lib/debug/usr/lib64/libQt5SerialBus.so.5.9.1.debug
--- /usr/lib/debug/usr/lib64/qt5/
--- /usr/lib/debug/usr/lib64/qt5/bin/
>>> /usr/lib/debug/usr/lib64/qt5/bin/canbusutil.debug
--- /usr/lib/debug/usr/lib64/qt5/plugins/
>>> /usr/lib/debug/usr/lib64/qt5/plugins/canbus/
>>> /usr/lib/debug/usr/lib64/qt5/plugins/canbus/libqttinycanbus.so.debug
>>> /usr/lib/debug/usr/lib64/qt5/plugins/canbus/libqtpeakcanbus.so.debug
>>> /usr/lib/debug/usr/lib64/qt5/plugins/canbus/libqtsocketcanbus.so.debug
>>> /usr/lib64/libQt5SerialBus.so -> libQt5SerialBus.so.5.9.1
>>> /usr/lib64/libQt5SerialBus.so.5 -> libQt5SerialBus.so.5.9.1
 * Regenerating gentoo-qconfig.h
 * Updating QT_CONFIG in qconfig.pri
>>> dev-qt/qtserialbus-5.9.1 merged.
>>> Regenerating /etc/ld.so.cache...

>>> Recording dev-qt/qtserialbus in "world" favorites file...
>>> Auto-cleaning packages...

>>> No outdated packages were found on your system.
``` 
The test was successfully performed with a Tiny-CAN IV-XL 4 USB CAN interface and Qt 5.9.1

### CODE REVIEW

File: dev-qt/qtserialbus/qtserialbus-5.9.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+if [[ ${QT5_BUILD_TYPE} == release ]]; then
+	KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~ppc64 ~x86"
+fi
+
+IUSE=""
+
+DEPEND="
+	~dev-qt/qtcore-${PV}
+    ~dev-qt/qtserialport-${PV}
```
Comment (Pesa) [inline dev-qt/qtserialbus/qtserialbus-5.9.1.ebuild]: fix indentation

File: dev-qt/qtserialbus/qtserialbus-5.9.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+"
+RDEPEND="${DEPEND}"
+
+src_prepare() {
+	qt5-build_src_prepare
```
Comment (Pesa) [inline dev-qt/qtserialbus/qtserialbus-5.9.1.ebuild]: this is redundant, you can avoid defining a `src_prepare` altogether.

File: dev-qt/qtserialbus/qtserialbus-5.9.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit qt5-build
+
+DESCRIPTION="Support for CAN and potentially other serial buses."
```
Comment (Pesa) [inline dev-qt/qtserialbus/qtserialbus-5.9.1.ebuild]: I think modbus is also supported?

delete "potentially" and don't end with a period

**Comment by kensington** [comment]
Thanks, I rebased your commits and picked them as 465542cd55d1b1e8abeae081aab71a70e86a79c1...8e3c208c9e16f43f9cb8c8c80826af053f697b7f.

**Pull request closed without merging**